antarest/study/business/output/variables_management.py

```diff
@@ -168,3 +172,209 @@ def extract_variables_list(output_path: Path) -> OutputVariablesList:
+    area_id: str | None = None,
+    area_from_id: str | None = None,
+    area_to_id: str | None = None,
+    thermal_id: str | None = None,
+    renewable_id: str | None = None,
+    st_storage_id: str | None = None,
+) -> OutputIdentifier:
+    output_identifier = _checks_variables_view_arguments_coherence(
+        variable_type, output_id, area_id, area_from_id, area_to_id, thermal_id, renewable_id, st_storage_id
+    )
+
+    if variable_type == OutputVariablesType.LINK:
+        assert isinstance(output_identifier, LinkOutputIdentifier)
+        _checks_links_variables_view_coherence(output_id, available_variables, variable_name, output_identifier)
+
+    else:
+        _checks_areas_variables_view_coherence(
+            output_id, available_variables, variable_name, output_identifier, variable_type
+        )
+
+    return output_identifier
+
+
+def _checks_links_variables_view_coherence(
+    output_id: str,
+    available_variables: OutputVariablesList,
+    variable_name: str,
+    output_identifier: LinkOutputIdentifier,
+) -> None:
+    area_from_id = output_identifier.area_from_id
+    area_to_id = output_identifier.area_to_id
+    error_msg = f"The variable '{variable_name}' does not exist for link '{area_from_id} - {area_to_id}'"
+    link_variables = available_variables.mc_ind.links
+    for link_variable in link_variables:
+        if link_variable.area_1_name == area_from_id and link_variable.area_2_name == area_to_id:
+            if variable_name in link_variable.variables:
+                return
+            raise OutputVariablesViewError(output_id, error_msg)
+
+    raise OutputVariablesViewError(output_id, error_msg)
+
+
+def _checks_areas_variables_view_coherence(
+    output_id: str,
+    available_variables: OutputVariablesList,
+    variable_name: str,
+    output_identifier: OutputIdentifier,
+    variable_type: OutputVariablesType,
+) -> None:
+    area_id = output_identifier.get_id_for_aggregation()
+    error_msg = f"The variable '{variable_name}' does not exist for area '{area_id}' and type '{variable_type.value}'"
+    area_variables = available_variables.mc_ind.areas
+    for area_variable in area_variables:
+        if area_variable.name == area_id:
+            match output_identifier:
+                case AreaOutputIdentifier():
+                    if variable_name in area_variable.variables:
+                        return
+                    raise OutputVariablesViewError(output_id, error_msg)
+                case ThermalClusterOutputIdentifier():
+                    attr = "thermal_clusters"
+                case RenewableClusterOutputIdentifier():
+                    attr = "renewable_clusters"
+                case ShortTermStorageOutputIdentifier():
+                    attr = "short_term_storages"
+                case _:
+                    raise OutputVariablesViewError(output_id, error_msg)
+            sub_id = output_identifier.get_sub_id_for_aggregation()
```

Comment:
we could name `variable` something like `cluster_variables`, in consistency with the class ?
it makes things a little difficult to follow to name it variable